lib/linguist/heuristics.rb

```diff
@@ -465,5 +465,13 @@ def call(data)
         Language["Scilab"]
       end
     end
+
+    disambiguate ".tsx" do |data|
+      if /^(import.+from|.+=\s*require)[\s\(]+['"]react['"]/.match(data)
+        Language["TypeScript"]
+      else
+        Language["XML"]
```

Comment:
Hmm, I guess I might be missing the nuance of how it would otherwise have worked previously (if not one or the other) 🤔

I can definitely add a condition for it, though 👍